gradio/helpers.py

```diff
@@ -138,24 +138,25 @@ def __init__(
                 "Please use gr.Examples(...) instead of gr.examples.Examples(...) to create the Examples.",
             )
 
+        self.cache_examples = False
         if cache_examples is None:
-            if cache_examples_env := os.getenv("GRADIO_CACHE_EXAMPLES"):
-                if cache_examples_env.lower() == "true":
-                    if fn is not None and outputs is not None:
-                        self.cache_examples = True
-                    else:
-                        self.cache_examples = False
-            elif utils.get_space() and fn is not None and outputs is not None:
+            if (
+                os.getenv("GRADIO_CACHE_EXAMPLES", "").lower() == "true"
+                and fn is not None
+                and outputs is not None
+            ):
                 self.cache_examples = True
-            else:
-                self.cache_examples = cache_examples or False
-        else:
-            if cache_examples not in [True, False]:
-                raise ValueError(
-                    "The `cache_examples` parameter must either: True or False."
-                )
+        elif cache_examples == "lazy":
+            warnings.warn(
+                "The `cache_examples` parameter no longer accepts a value of 'lazy'. To enable lazy caching in "
```

Comment:
Why not enable lazy caching in this case instead of not caching at all since we're soft deprecating?

Reply:
Yes that's fair actually, I'll make the change